app/src/main/java/app/revanced/integrations/youtube/patches/components/KeywordContentFilter.java

```diff
@@ -0,0 +1,156 @@
+package app.revanced.integrations.youtube.patches.components;
+
+import static app.revanced.integrations.shared.StringRef.str;
+import static app.revanced.integrations.youtube.ByteTrieSearch.convertStringsToBytes;
+
+import android.os.Build;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.RequiresApi;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+import app.revanced.integrations.shared.Logger;
+import app.revanced.integrations.shared.Utils;
+import app.revanced.integrations.youtube.ByteTrieSearch;
+import app.revanced.integrations.youtube.settings.Settings;
+
+/**
+ * <pre>
+ * Allows hiding home feed and search results based on keywords and/or channel names.
+ *
+ * Limitations:
+ * - Searching for a keyword phrase will give no search results
+ * - Filtering a channel name can still show Shorts from that channel in the search results
+ * - Some layout component residue will remain, such as the video chapter previews for some search results.
+ * - Keywords are case sensitive, but some casing variation is manually added.
+ *   (ie: "mr beast" automatically filters "Mr Beast" and "MR BEAST").
+ */
+@SuppressWarnings("unused")
+@RequiresApi(api = Build.VERSION_CODES.N)
+final class KeywordContentFilter extends Filter {
+
+    /**
+     * Minimum keyword/phrase length to prevent excessively broad content filtering.
+     */
+    private static final int MINIMUM_KEYWORD_LENGTH = 3;
+
+    /**
+     * Substrings that are always first in the path.
+     */
+    private final StringFilterGroup startsWithFilter = new StringFilterGroup(
+            Settings.HIDE_KEYWORD_CONTENT,
+            "home_video_with_content.eml",
+            "search_video_with_context.eml",
+            "related_video_with_content",
+            "compact_video.eml",
+            "inline_shorts",
```

Comment:
compact_playlist.eml seems to be present under a channels "Playlist" tab. I think it could also appear under other feeds:

![image](https://github.com/ReVanced/revanced-integrations/assets/13122796/981819e2-d078-4fa7-898e-6ed6faee7a08)

Reply:
compact_playlist are the playlist components visible in the screenshot I attached:

![image](https://github.com/ReVanced/revanced-integrations/assets/13122796/7762af0e-bba5-43c6-83f8-ee4069c449f0)

Reply:
There are "Mix" playlists in feeds, but I am not sure if those are the same components. Unfortunately, I can not get them to appear on my feeds, but I can imagine that "compact_playlist" could appear in the feeds, just like "compact_video"